System/etc/profile.d/nextspace.sh

```diff
@@ -5,8 +5,8 @@ export PATH=$NS_PATH:$PATH
 
 export MANPATH=:/Library/Documentation/man:/usr/NextSpace/Documentation/man
 
-### Moved into NSGlobalDomain prefs. Format 'Local Time Zone' = Europe/Kiev
-#export GNUSTEP_TZ=`cat /etc/timezone`
+# Initialize time zone to system time zone, but only if not yet set.
+defaults write $(defaults read NSGlobalDomain 'Local Time Zone' 2>/dev/null || echo "NSGlobalDomain 'Local Time Zone' $(cat /etc/timezone)")
```

Comment:
RHEL/CentOS doesn't have `/etc/timezone` file. I guess we need different method to determine system configured timezone.

Reply:
Completely untested but the following should work on systemd-equipped and Debian-style systems: `(timedatectl show -p Timezone --value || cat /etc/timezone) 2>/dev/null`

It certainly emits the right timezone for me on Debian. If you could verify on RHEL/CentOS and agree that this could work, how about we do that?

Reply:
Unfortunaltely `timedatectl` has no `show` option on CentOS 7. CentOS 8 - OK. I guess the appropriate soltion of CentOS 7 is parsing /etc/localtime link destination and extract time zone value from it. My localtime is:
```
$ ls -l /etc/localtime
lrwxrwxrwx. 1 root root 33 Jun  9 19:44 /etc/localtime -> ../usr/share/zoneinfo/Europe/Kiev
```

Reply:
`/usr/bin/env stat --printf "%N\n" /etc/localtime |sed "s,.*-> '.*/zoneinfo/\([^']*\)',\1,"` works on a centos docker instance I spun up. Can you confirm?

With that we can do `(timedatectl show -p Timezone --value || /usr/bin/env stat --printf "%N\n" /etc/localtime |sed "s,.*-> '.*/zoneinfo/\([^']*\)',\1,")`

Reply:
It returns `‘/etc/localtime’ -> ‘../usr/share/zoneinfo/Europe/Kiev’` for me. What is your output? Something wrong with `sed` expression?

Reply:
Looking around for a solution I found a few reports that localtime isn't even a symlink on some devices. The Redhat way of configuring timezones is to set ZONE in /etc/sysconfig/clock, so maybe `(timedatectl show -p Timezone --value || (source /etc/sysconfig/clock && echo $ZONE) || cat /etc/timezone || echo UTC) 2>/dev/null`?

It tries timedatectl first (because that's the modern approach), looks for sysconfig/clock next (but since my centos 7 container doesn't have the file I'm not sure if that's the right way to parse it, but from examples on the net that should work), checks out /etc/timezone and if everything fails returns UTC as a fallback default.

Since all that stuff is only executed when there's not already a timezone configured in the defaults database, the runtime impact shouldn't be too bad.

Reply:
I don't know what reports you're talking about (maybe versions 5 or 6?) but since version 7 of RHEL/CentOS:
1. /etc/localtime is a symlink to /usr/share/zoneinfo subdirectory.
2. There is no /etc/sysconfig/clock.

Trying to understand why your `sed` export doesn't work I've found some helpful things:
- LC_CTYPE value affects `sed` results (it's expected): default value is `UTF-8` and your `/usr/bin/env stat --printf "%N\n" /etc/localtime |sed "s,.*-> '.*/zoneinfo/\([^']*\)',\1,"` works like a charm;
- with default value of LC_CTYPE some programs complains about locale setup; for example here the `man` output: `man: can't set the locale; make sure $LC_* and $LANG are correct`.

Finally in original file line with `export LC_CTYPE` should be changed to `export LC_CTYPE=$LANG` (I hope this is a correct setup) and timezone setup line should use `/usr/bin/env -i stat --printf "%N\n" /etc/localtime |sed "s,.*-> '.*/zoneinfo/\([^']*\)',\1,"` (note `-i` option  I've added).

Reply:
Here is my variant:
```shell
# NSGlobalDomain preference. Format 'Local Time Zone' = Europe/Kiev
defaults read NSGlobalDomain "Local Time Zone" 2>&1 > /dev/null
if [ $? -ne 0 ]; then
    echo "Updating `Local Time Zone` preference..."
    TZ=`/usr/bin/env -i stat --printf "%N\n" /etc/localtime |sed "s,.*-> '.*/zoneinfo/\([^']*\)',\1,"`
    defaults write NSGlobalDomain "Local Time Zone" $TZ
fi
```
What do you think?